tests/bluetooth/tester/src/btp_vcp.c

```diff
@@ -0,0 +1,544 @@
+	int err;
+
+	LOG_DBG("Volume Up");
+
+	err = bt_vcp_vol_rend_vol_up();
+	if (err) {
+		status = BTP_STATUS_FAILED;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_VOL_UP, CONTROLLER_INDEX, status);
+}
+
+static void vol_down(void)
+{
+	uint8_t status = BTP_STATUS_SUCCESS;
+	int err;
+
+	LOG_DBG("Volume Down");
+
+	err = bt_vcp_vol_rend_vol_down();
+	if (err) {
+		status = BTP_STATUS_FAILED;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_VOL_DOWN, CONTROLLER_INDEX, status);
+}
+
+static void mute(void)
+{
+	uint8_t status = BTP_STATUS_SUCCESS;
+	int err;
+
+	err = bt_vcp_vol_rend_mute();
+	if (err) {
+		status = BTP_STATUS_FAILED;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_MUTE, CONTROLLER_INDEX, status);
+}
+
+static void unmute(void)
+{
+	uint8_t status = BTP_STATUS_SUCCESS;
+	int err;
+
+	err = bt_vcp_vol_rend_unmute();
+	if (err) {
+		status = BTP_STATUS_FAILED;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_UNMUTE, CONTROLLER_INDEX, status);
+}
+
+static void vcs_state_cb(int err, uint8_t volume, uint8_t mute)
+{
+	if (err) {
+		LOG_DBG("VCP state cb err (%d)", err);
+		return;
+	}
+}
+
+static void vcs_flags_cb(int err, uint8_t flags)
+{
+	if (err) {
+		LOG_DBG("VCP flags cb err (%d)", err);
+		return;
+	}
+}
+
+static struct bt_vcp_vol_rend_cb vcs_cb = {
+	.state = vcs_state_cb,
+	.flags = vcs_flags_cb,
+};
+
+static void vcp_init(void)
+{
+	struct bt_vcp_vol_rend_register_param vcp_register_param;
+	int err;
+
+	/* Default values */
+	vcp_register_param.step = 1;
+	vcp_register_param.mute = BT_VCP_STATE_UNMUTED;
+	vcp_register_param.volume = 100;
+	vcp_register_param.cb = &vcs_cb;
+
+	LOG_DBG("");
+
+	err = bt_vcp_vol_rend_register(&vcp_register_param);
+	if (err) {
+		tester_rsp(BTP_SERVICE_ID_VCS, VCS_INIT, CONTROLLER_INDEX,
+			   BTP_STATUS_FAILED);
+	}
+}
+
+void tester_handle_vcs(uint8_t opcode, uint8_t index, uint8_t *data,
+		       uint16_t len)
+{
+	switch (opcode) {
+	case VCS_READ_SUPPORTED_COMMANDS:
+		vcs_supported_commands(data, len);
+		break;
+	case VCS_INIT:
+		vcp_init();
+		break;
+	case VCS_SET_VOL:
+		set_volume(data, len);
+		break;
+	case VCS_VOL_UP:
+		vol_up();
+		break;
+	case VCS_VOL_DOWN:
+		vol_down();
+		break;
+	case VCS_MUTE:
+		mute();
+		break;
+	case VCS_UNMUTE:
+		unmute();
+		break;
+	default:
+		tester_rsp(BTP_SERVICE_ID_VCS, opcode, index,
+			   BTP_STATUS_UNKNOWN_CMD);
+		break;
+	}
+}
+
+/* Audio Input Control Service */
+static void aics_supported_commands(uint8_t *data, uint16_t len)
+{
+	struct net_buf_simple *buf = NET_BUF_SIMPLE(BTP_DATA_MAX_SIZE);
+
+	net_buf_simple_init(buf, 0);
+	net_buf_simple_add_u8(buf, AICS_READ_SUPPORTED_COMMANDS);
+	net_buf_simple_add_u8(buf, AICS_INIT);
+	net_buf_simple_add_u8(buf, AICS_SET_GAIN);
+	net_buf_simple_add_u8(buf, AICS_MUTE);
+	net_buf_simple_add_u8(buf, AICS_UNMUTE);
+	net_buf_simple_add_u8(buf, AICS_MAN_GAIN);
+	net_buf_simple_add_u8(buf, AICS_AUTO_GAIN);
+	net_buf_simple_add_u8(buf, AICS_MAN_GAIN_ONLY);
+	net_buf_simple_add_u8(buf, AICS_AUTO_GAIN_ONLY);
```

Comment:
Do we really have both of these commands supported?

Reply:
Nope, removed